src/main/java/org/gridsuite/modification/server/modifications/GenerationDispatch.java

```diff
@@ -522,13 +523,65 @@ public void apply(Network network, Reporter subReporter) {
             Reporter resultReporter = componentReporter.createSubReporter(RESULT, RESULT);
 
             if (Math.abs(totalAmountSupplyToBeDispatched - realized) < EPSILON) {
+                Map<String, List<Generator>> generatorsByRegion = getGeneratorsByRegion(network);
+
                 report(resultReporter, Integer.toString(componentNum), "SupplyDemandBalanceCouldBeMet", "The supply-demand balance could be met",
                     Map.of(), TypedValue.INFO_SEVERITY);
+                generatorsByRegion.forEach((region, generators) -> report(resultReporter, Integer.toString(componentNum), "SumGeneratorActivePower" + region, "Sum of generator active power setpoints in ${region} region: ${sum} MW.",
+                        Map.of("region", region, "sum", getSumActivePower(generators)), TypedValue.INFO_SEVERITY));
             } else {
                 double remainingPowerImbalance = totalAmountSupplyToBeDispatched - realized;
                 report(resultReporter, Integer.toString(componentNum), "SupplyDemandBalanceCouldNotBeMet", "The supply-demand balance could not be met : the remaining power imbalance is ${remainingPower} MW",
                     Map.of("remainingPower", remainingPowerImbalance), TypedValue.WARN_SEVERITY);
             }
         }
     }
+
+    private Map<String, List<Generator>> getGeneratorsByRegion(Network network) {
+        // get all connected generators and the substationIds associated.
+        List<Generator> connectedGenerators = network.getGeneratorStream()
+                .filter(g -> g.getTerminal().isConnected())
+                .toList();
+        List<String> substationIds = connectedGenerators.stream()
+                .map(g -> g.getTerminal().getVoltageLevel().getSubstation().orElseThrow().getId())
```

Comment:
Don't throw an exception for a voltageLevel not in a substation, which can occur : just ignore the generator in that case

Reply:
Done.